src/property-filter/token-editor-inputs.tsx

```diff
@@ -168,3 +181,42 @@ export function ValueInput({
     />
   );
 }
+
+interface ValueInputPropsEnum extends ValueInputProps {
+  property: InternalFilteringProperty;
+  operator: ComparisonOperator;
+}
+
+function ValueInputEnum({
+  property,
+  operator,
+  value: unknownValue,
+  onChangeValue,
+  asyncProps,
+  filteringOptions,
+  onLoadItems,
+}: ValueInputPropsEnum) {
+  const valueOptions = filteringOptions
+    .filter(option => option.property?.propertyKey === property.propertyKey)
+    .map(({ label, value }) => ({ label, value }));
+  const valueAutosuggestHandlers = useLoadItems(onLoadItems, '', property.externalProperty, undefined, operator);
+  const asyncValueAutosuggestProps = { statusType: 'finished' as const, ...valueAutosuggestHandlers, ...asyncProps };
+  const value = Array.isArray(unknownValue) ? unknownValue : [unknownValue];
```

Comment:
Why do we check against `!unknownValue` [here](https://github.com/cloudscape-design/components/pull/2739/files#diff-20f3d037528e34ddb0726f12a7de9c8adf043d278644c88d29c4485259d0a605R81) in `property-editor.tsx` but here we don't?

Reply:
Good call, I will update!

In the token editor we generally ensure the temp value for enum tokens is created as an empty array instead of null, however it can still be null when coming from the customer-provided query token.